### DISCUSSION TITLE
Corrupted Database

### DISCUSSION THREAD

**Discussion by KingofAnomaly** [plan-player-analytics/Plan#3817] [Q&A]
My database got corrupted and I have no idea how to fix it without losing all my data. Is there any way to save the data?

**Comment by AuroraLS3** [comment]
Which database type?

**Reply by KingofAnomaly** [reply]
SQLite

**Comment by AuroraLS3** [comment]
SQLITE:

- Install sqlite3 commandline tools https://www.digitalocean.com/community/tutorials/how-to-install-and-use-sqlite-on-ubuntu-20-04
- `sqlite3 broken.db ".recover" | sqlite3 new.db`

**Reply by KingofAnomaly** [reply]
Thank you I'll try this when I get home from work.

**Reply by KingofAnomaly** [reply]
I did the command exactly as you said but it gives me this 
"Error: unknown command or invalid arguments:  "recover". Enter ".help" for help"
not sure if using a mac makes a difference or not

**Reply by AuroraLS3** [reply]
You need to put `".recover"` instead of "recover"

**Reply by KingofAnomaly** [reply]
Yeah I typed it exactly as you had it so I put ".recover" is it possible I have a very old version of it and is it something I can update? And the whole command I input was sqlite3 broken.db ".recover" | sqlite3 new.db and I put my database in the same folder as the tool file and renamed it.

**Reply by AuroraLS3** [reply]
Yes I think it's a very recent new feature

**Reply by KingofAnomaly** [reply]
not sure if it's possible for you to fix it because I can't figure out if the version I installed is old or if it's just using a version that was already on my computer. https://drive.google.com/file/d/19_HQp2Z_e4jDOcRR93fk3jR4M8LCVLgK/view?usp=drive_link

**Reply by AuroraLS3** [reply]
I don't feel comfortable sharing my google email to access the drive file

**Comment by AuroraLS3** [comment]
MYSQL:

- https://www.digitalocean.com/community/tutorials/how-to-fix-corrupted-tables-in-mysql
